src/main/java/org/folio/inventory/dataimport/handlers/actions/UpdateHoldingEventHandler.java

```diff
@@ -201,9 +207,31 @@ private void prepareDataAndReInvokeCurrentHandler(DataImportEventPayload dataImp
     });
   }
 
-  private void constructDataImportEventPayload(CompletableFuture<DataImportEventPayload> future, DataImportEventPayload dataImportEventPayload, HoldingsRecord holding) {
-    dataImportEventPayload.getContext().put(HOLDINGS.value(), Json.encodePrettily(holding));
-    future.complete(dataImportEventPayload);
-    future.complete(dataImportEventPayload);
+  private void constructDataImportEventPayload(CompletableFuture<DataImportEventPayload> future, DataImportEventPayload dataImportEventPayload, HoldingsRecord holding, Context context) {
+    HashMap<String, String> payloadContext = dataImportEventPayload.getContext();
+    payloadContext.put(HOLDINGS.value(), Json.encodePrettily(holding));
+    if(payloadContext.containsKey(ITEM.value())) {
+      ItemCollection itemCollection = storage.getItemCollection(context);
+      updateDataImportEventPayloadItem(future, dataImportEventPayload, itemCollection);
+    } else {
+      future.complete(dataImportEventPayload);
+    }
+  }
+
+  private void updateDataImportEventPayloadItem(CompletableFuture<DataImportEventPayload> future, DataImportEventPayload dataImportEventPayload, ItemCollection itemCollection) {
+    JsonObject oldItemAsJson = new JsonObject(dataImportEventPayload.getContext().get(ITEM.value()));
+    String itemId = oldItemAsJson.getString(ITEM_ID_HEADER);
+    itemCollection.findById(itemId, findResult -> {
+      if(Objects.nonNull(findResult)) {
+        JsonObject itemAsJson = new JsonObject(ItemUtil.mapToMappingResultRepresentation(findResult.getResult()));
+        dataImportEventPayload.getContext().put(ITEM.value(), Json.encode(itemAsJson));
+      }
+      future.complete(dataImportEventPayload);
+    }, failure -> {
+      EventProcessingException processingException =
+        new EventProcessingException(format(CANNOT_GET_ACTUAL_ITEM_MESSAGE, itemId, failure.getReason(), failure.getStatusCode()));
```

Comment:
@Aliaksandr-Fedasiuk  String format uses %s as placeholder, not {}.
{} palceholder used in logs

Reply:
fixed. Thank you, @SerhiiNosko!